core/src/components/nav/nav.tsx

```diff
@@ -872,14 +872,15 @@ export class Nav implements NavOutlet {
       mode,
       showGoBack: this.canGoBackSync(enteringView),
       baseEl: this.el,
-      animationBuilder: this.animation || opts.animationBuilder || config.get('navAnimation'),
       progressCallback,
       animated: this.animated && config.getBoolean('animated', true),
 
       enteringEl,
       leavingEl,
 
-      ...opts
+      ...opts,
+
+      animationBuilder: opts.animationBuilder || this.animation || config.get('navAnimation')
```

Comment:
Any reason why we switched the order here?

Reply:
Transition passed in opts (e.g. it came from router) should take precedence over nav transition. Simple example - I set transition for nav, but in specific scenario, when moving from view A to B I need other transition, which can be passed to push() or set in router-link. Previous impl would always pick nav's transition.

Reply:
Good catch! In that case, could we also make that same change here: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/router-outlet/route-outlet.tsx#L191

Reply:
Done